pkg/pillar/cmd/zedagent/parseconfig.go

```diff
@@ -632,6 +632,11 @@ func parseAppInstanceConfig(getconfigCtx *getconfigContext,
 		appInstance.FixedResources.DisableLogs = cfgApp.Fixedresources.DisableLogs
 		appInstance.MetaDataType = types.MetaDataType(cfgApp.MetaDataType)
 		appInstance.Delay = time.Duration(cfgApp.StartDelayInSeconds) * time.Second
+		appInstance.Service = cfgApp.Service
+		if appInstance.Service && appInstance.FixedResources.VirtualizationMode != types.NOHYPER {
+			log.Warnf("MobyLabel ConfigMode app instance %s should have NOHYPER VirtualizationMode", appInstance.UUIDandVersion.UUID)
+			appInstance.FixedResources.VirtualizationMode = types.NOHYPER
+		}
```

Comment:
Can we make this return an error instead of just logging a warning? That would be easier if the check is moved from zedagent to zedmanager or domainmgr unless parseAppInstanceConfig() already has the structure to add parse errors to appInstanceConfig (which zedmanager turns into errors in AppInstanceStatus). (We do have such a structure for some other objects in parseconfig.go but I'm not sure about AppInstanceConfig.)

Reply:
Make sense, thank you! And we have `checkAndPublishAppInstanceConfig` which is perfectly fit.